container-core/src/main/java/com/yahoo/container/core/config/HandlersConfigurerDi.java

```diff
@@ -138,11 +138,18 @@ public void useBundles(Collection<FileReference> bundles) {
         }
     }
 
-    public void runOnceAndEnsureRegistryHackRun(Injector discInjector) throws InterruptedException {
-        currentGraph = container.runOnce(currentGraph, createFallbackInjector(vespaContainer, discInjector));
-
-        RegistriesHack registriesHack = currentGraph.getInstance(RegistriesHack.class);
-        assert (registriesHack != null);
+    /**
+     * Wait for new config to arrive and produce the new graph
+     *
+     * @return true if this resulted in a new graph that should be applied to the currently running container
+     */
+    public boolean getNewConfigGraph(Injector discInjector, boolean restartOnRedeploy) throws InterruptedException {
```

Comment:
You probably mean ComponentGraph instead of ConfigGraph

Reply:
Hm, yeah but I also wanted to say "get config" ... but ok I'll rename.